cmd/warm-backend-gcs.go

```diff
@@ -31,7 +32,7 @@ func (gcs *warmBackendGCS) Put(ctx context.Context, key string, data io.Reader,
 		w.ObjectAttrs.StorageClass = gcs.StorageClass
 	}
 	if _, err := io.Copy(w, data); err != nil {
-		return err
+		return gcsToObjectError(err, gcs.Bucket, key)
```

Comment:
you can return just `err` here since gcsToObjectError() is used for returning S3 compatible errors to S3 client

Reply:
Or this is fine, we can be uniform in behavior of returning ObjectError from warm-backend layer